src/sgame/botlib/bot_nav_edit.cpp

```diff
@@ -276,6 +282,96 @@ void Cmd_AddConnection( gentity_t *ent )
 
 	arg = Cmd_Argv( 1 );
 
+	if ( cmd.enabled )
```

Comment:
Instead of copy and pasting the whole function, I suggest using the same function to implement both commands, and switching the point placement behavior based on Cmd_Argv(0).

Reply:
> Can you write a bit of documentation on how to use the addcon stuff somewhere? I think you're the only person who knows how to use it, and understanding what it's for would help in reviewing the commit.

There is documentation in 2 places, currently:

* a small tutorial [https://forums.unvanquished.net/viewtopic.php?f=8&t=2244&sid=93e4d95f4a588299c789cd597e1acd12#p18577](here)
* a wiki page [https://wiki.unvanquished.net/wiki/Navigation_Meshes](there)

I agree that wiki page's quality can be improved, and that a better doc could be written, and that's even something I'd like to do at some point... *when* I will understand enough the navedit commands, as there are commands, notably, to somehow force bots to go somewhere.

With my current knowledge of the problem, I can't really write a good enough doc imo. There's also likely a «tips and tricks» database to build, and that kind of things, which even depends on what features are around.
For example, my code to allow the use of ladders and wallwalking is not ready yet, and might allow to start such a database.
At some time, I will want to implement "reach by jump or pounce" code, so that bots can access disconnected areas which are _above_ their current one. This will likely require changing the API, and thus rewrite the doc.

What can likely be done, though, is to open a ticket for next release to write/update the doc for this. I think this would be a good thing to do that for each releases: take time to update stuff which requires documentation, especially for user-facing feature. That's a different topic though.

Reply:
Alright. Let's use the occasion to improve it further, then.

Reply:
Change of mind.
I don't want to break previous API, and handling optional parameters in this stuff is super tedious for nothing, it's even more painful than using getopt.

I will rewrite those API cleanly, yes. But this will happen *when* I'll know how to annotate the navcons with the method required to use it (ladder, WW, jump with req magnitude, pounce with magnitude, wall jump, maybe fly, etc).

For now, it's really just lot of very tedious work to do stuff correctly for no benefit. I'll still add a TODO note with what is planned, though.